app/src/main/java/com/example/a310_rondayview/ui/InterestedEventsAdapter.java

```diff
@@ -40,6 +47,42 @@ public void onBindViewHolder(@androidx.annotation.NonNull InterestedEventsViewHo
         // Binds data to the UI elements in each list item
         Event event = eventsList.get(position);
 
+        ScaleAnimation scaleAnimation = new ScaleAnimation(0.7f, 1.0f, 0.7f, 1.0f, Animation.RELATIVE_TO_SELF, 0.7f, Animation.RELATIVE_TO_SELF, 0.7f);
+        scaleAnimation.setDuration(500);
+        BounceInterpolator bounceInterpolator = new BounceInterpolator();
+        scaleAnimation.setInterpolator(bounceInterpolator);
+
+        ToggleButton heartButton = holder.itemView.findViewById(R.id.heart_button);
+
+        heartButton.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(CompoundButton compoundButton, boolean isChecked) {
+
+                Animation fadeOut = AnimationUtils.loadAnimation(context, R.anim.fade_out);
+                holder.itemView.startAnimation(fadeOut);
+
+                // Delay the removal of the item to match the animation duration
+                // Code snippet adapted from CHAT-GPT
```

Comment:
great that you refferenced this code snippet from ChatGPT, if you could just check out this link, it shows you syntax for refferencing AI, should be a quick fix to update the referencing but thank you for your honesty

Reply:
https://www.cite.auckland.ac.nz/2_8_22.html